idioms/priv-extend.md

```diff
@@ -1,46 +1,103 @@
-# Privacy for extensibility
+# `#[non_exhaustive]` & private fields for extensibility
 
 ## Description
 
-Use a private field to ensure that a struct is extensible without breaking
-stability guarantees.
+A small set of scenarios exist where a library author may want to add public
+fields to a public struct or new variants to an enum without breaking backwards
+compatibility. Rust offers two solutions:
+
+- Use `#[non_exhaustive]` on `struct`s, `enum`s, and `enum` variants.
+For extensive documentation on all the places `#[non_exhaustive]` can be used,
+see [the docs](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute).
+
+- For `struct`s only, you may add a private field to struct
```

Comment:
That's not entirely true.
If before you have:
```rust
struct Point {
    pub x: u32,
    pub y: u32
}
```

and than you do:
```rust
struct Point {
    pub x: u32,
    pub y: u32,
    z: u32
}
```

This is not backward compatible, because previous code like `let a = Point {x:1, y:2}` won't work anymore

Reply:
You can add something like "if other private fields already existed" to the sentence in order to fix this.
In fact if other private fields existed, you were not already allowed to create a point like that.